### PR TITLE
Update the support for lazydesigners/dimple

### PR DESCRIPTION
**Pull request by jackytrabbit**
Update the support for lazydesigners/dimple

## Description

Update the support for lazydesigners/dimple
Add the support for new layouts of Dimple round 2

## Types of Changes

- [ ] Core
- [ ] Bugfix
- [ ] New feature
- [ ] Enhancement/optimization
- [x] Keyboard (addition or update)
- [ ] Keymap/layout/userspace (addition or update)
- [ ] Documentation

## Issues Fixed or Closed by This PR

* 

## Checklist

- [x] My code follows the code style of this project: [**C**](https://docs.qmk.fm/#/coding_conventions_c), [**Python**](https://docs.qmk.fm/#/coding_conventions_python)
- [x] I have read the [**PR Checklist** document](https://docs.qmk.fm/#/pr_checklist) and have made the appropriate changes.
- [x] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have read the [**CONTRIBUTING** document](https://docs.qmk.fm/#/contributing).
- [x] I have added tests to cover my changes.
- [x] I have tested the changes and verified that they work and don't break anything (as well as I can manage).

### CODE REVIEW

File: keyboards/lazydesigners/dimple/staggered/keymaps/oncesavedgaming/keymap.c [hunk 1]
```diff
@@ -0,0 +1,7 @@
+#include QMK_KEYBOARD_H
```
Comment (drashna) [inline keyboards/lazydesigners/dimple/staggered/keymaps/oncesavedgaming/keymap.c:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

File: keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c [hunk 2]
```diff
@@ -0,0 +1,53 @@
...
+ * |------------------------------------------------------------------|
+ * |Tab/LT2| A  | S  | D  | F  | G  | H  | J  | K  | L  | '  |  |Enter|
+ * |------------------------------------------------------------------|
+ * | Shift | Z  | X  | C  | V  | B  | N  | M  | ,  | .  | Up | MO(2)  |
+ * |------------------------------------------------------------------|
+ *      |Ctrl|Gui |Alt | Spc/LT1  | MO(3)|   Spc   |Left|Down|Rght|
+ *      `---------------------------------------------------------'
+ */
+
+ const uint16_t PROGMEM keymaps[][MATRIX_ROWS][MATRIX_COLS] = {
```
Comment (drashna) [inline keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c:34]: This should have 4 layers

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c (hunk 2 above)
Comment (jackytrabbit) [reply keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c:34]: Sorry about that. 'MO(3)' was a mistake.

File: keyboards/lazydesigners/dimple/staggered/keymaps/oncesavedgaming/keymap.c (hunk 1 above)
Comment (jackytrabbit) [reply keyboards/lazydesigners/dimple/staggered/keymaps/oncesavedgaming/keymap.c:17]: Thank you! License header is added.

File: keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c (hunk 2 above)
Comment (drashna) [reply keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c:34]: Regardless, this still needs to have 4 layers. 

By default, VIA assumes 4 layers, and if they're not all populated, then the unpopulated layers will be filled with random garbage.  

So you need to either have 4 layers, or you need to set the define that limits it to 3 layers.

File: keyboards/lazydesigners/dimple/staggered/keymaps/erovia/config.h [hunk 3]
```diff
@@ -0,0 +1 @@
+#define TAPPING_TERM 170
```
Comment (drashna) [inline keyboards/lazydesigners/dimple/staggered/keymaps/erovia/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#pragma once

#define TAPPING_TERM 170
```

File: keyboards/lazydesigners/dimple/staggered/keymaps/erovia/keymap.c [hunk 4]
```diff
@@ -0,0 +1,162 @@
...
+
+	[_VIM] = LAYOUT(
+		EEP_RST, RESET,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,
+		KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_H,    KC_J,    KC_K,    KC_L,    KC_NO,   KC_NO,
+		KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_NO,   KC_UP,   KC_NO,
+		RGB_VAD, RGB_TOG, RGB_VAI, KC_NO,   KC_NO,   KC_LEFT, KC_DOWN, KC_RGHT
+	),
+};
+
+uint32_t layer_state_set_keymap(uint32_t state) {
```
Comment (drashna) [inline keyboards/lazydesigners/dimple/staggered/keymaps/erovia/keymap.c:141]:
```suggestion
layer_state_t layer_state_set_keymap(layer_state_t state) {
```

File: keyboards/lazydesigners/dimple/staggered/rev1/config.h [hunk 5]
```diff
@@ -0,0 +1,2 @@
+#define LED_CAPS_LOCK_PIN E6
```
Comment (drashna) [inline keyboards/lazydesigners/dimple/staggered/rev1/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#pragma once

#define LED_CAPS_LOCK_PIN E6
```

File: keyboards/lazydesigners/dimple/staggered/rev3/config.h [hunk 6]
```diff
@@ -0,0 +1,4 @@
+#define BACKLIGHT_PIN B7
```
Comment (drashna) [inline keyboards/lazydesigners/dimple/staggered/rev3/config.h:1]: Could you add a GPL2+ compatible license header here? 

For instance: 
https://github.com/qmk/qmk_firmware/blob/28ff51175b6d45ce5a8dbfe7ed6e7a12df3bc8a8/quantum/template/base/keyboard.c#L1-L15

```suggestion
#pragma once

#define BACKLIGHT_PIN B7
```

**Review by drashna — CHANGES_REQUESTED** [review]

File: keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c (hunk 2 above)
Comment (jackytrabbit) [reply keyboards/lazydesigners/dimple/ortho/keymaps/via/keymap.c:34]: I see. Thank you!

**Comment by jackytrabbit** [comment]
@tominabox1 Need a help here.

File: keyboards/lazydesigners/dimple/staggered/staggered.c [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+#include "staggered.h"
```
Comment (tominabox1) [inline keyboards/lazydesigners/dimple/staggered/staggered.c:16]:
```SUGGESTION
#include "staggered.h"
void dimple_led_on() {
    writePinLow(E6);
  }

 void dimple_led_off() {
    writePinHigh(E6);
  }
```

Fixes build errors

File: keyboards/lazydesigners/dimple/staggered/staggered.h [hunk 8]
```diff
@@ -40,7 +40,3 @@
 	{ K200,  K201,  K202,  K203,  K204,  K205,  K206,  K207,  K208,  K209,  K210,  KC_NO }, \
 	{ K300,  KC_NO, K302,  K303,  K304,  KC_NO, K306,  K307,  K308,  K309,  KC_NO, KC_NO }  \
 }
```
Comment (tominabox1) [inline keyboards/lazydesigners/dimple/staggered/staggered.h:42]:
```SUGGESTION
}
void dimple_led_on(void);
void dimple_led_off(void);
```

with above, fixes build errors

**Review by tominabox1 — CHANGES_REQUESTED** [review]
These changes should fix the build errors you're having, minus erovia's issue with missing .h file. Also one quality of life adjustment.

**Comment by jackytrabbit** [comment]
> These changes should fix the build errors you're having, minus erovia's issue with missing .h file. Also one quality of life adjustment.

Thank you so much! It works.

**Review by drashna — APPROVED** [review]

File: keyboards/lazydesigners/dimple/staggered/rev3/info.json [hunk 9]
```diff
@@ -0,0 +1,146 @@
+{
+    "keyboard_name": "Dimple", 
+    "url": "http://lazydesigners.cn", 
+    "maintainer": "Jacky@LAZYDESIGNERS", 
+    "width": 12.5, 
+    "height": 4,
```
Comment (fauxpark) [inline keyboards/lazydesigners/dimple/staggered/rev3/info.json:6]:
```suggestion
```

**Review by fauxpark — CHANGES_REQUESTED** [review]
Permissions changes need to be reverted.
image

**Comment by jackytrabbit** [comment]
> Permissions changes need to be reverted.
> image

Is there any thing I should do with this?

**Comment by fauxpark** [comment]
https://stackoverflow.com/questions/10516201/updating-and-committing-only-a-files-permissions-using-git-version-control

**Comment by jackytrabbit** [comment]
> https://stackoverflow.com/questions/10516201/updating-and-committing-only-a-files-permissions-using-git-version-control

@fauxpark Thank you so much!

**Comment by fauxpark** [comment]
Looks good, except for the merge conflict.

**Pull request closed without merging**